### ISSUE TITLE
Metamask freezes when trying to confirm transaction on Trezor T

### ISSUE THREAD

**Issue by Fefiox** [MetaMask/metamask-extension#12967]
When trying to confirm the transaction after giving access to metamask to my trezor T wallet, it freezes and doesn´t continue, freezing also the browser.

**Steps to reproduce (REQUIRED)**
Steps to reproduce the behavior, libraries used with version number, and/or any setup information to easily reproduce:

1. Make a transaction
2. Opens a new page where metamask asks for permission to confirm the transaction on the Trezor T
3. Give access
4. When the page closes and the transaction should appear to confirm on the Trezor, nothing happens and the app freezes.

**Expected behavior**
Confirm the transaction on the Trezor T.

**Browser details (please complete the following information):**
 - OS: Windows
 - Hardware Wallet: Trezor Firmware version 2.4.2
 - Browser: Firefox
 - MetaMask Version:10.6.4

**Comment by danjm** [comment]
Thanks for the report. Was this on the ethereum mainnet network?

**Comment by Fefiox** [comment]
It was on Avalanche mainnet, but it happens in all of them.

**Comment by prusnak** [comment]
This is most probably a duplicate of https://github.com/MetaMask/metamask-extension/issues/12130

**Comment by Sukha5911** [comment]
U uthe

**Comment by Einoow** [comment]
Same problem, waiting for fix.

**Comment by Einoow** [comment]
Hey! Any update about this problem?

Browser details (please complete the following information):

OS: Windows
Hardware Wallet: Trezor model T Firmware version 1.10.3
Browser: Firefox
MetaMask Version:10.8.0

**Comment by prusnak** [comment]
> Hardware Wallet: Trezor model one Firmware version 1.10.3

@Einoow the issue has been fixed in Metamask 10.8.0 but you need to update your firmware to version 1.10.4

**Comment by Einoow** [comment]
Hey! Thanks for your response, I have made some tests and seems like the problem is still there for me. 

OS: Windows
Hardware Wallet: Trezor model one Firmware version 1.10.4
Browser: Firefox
MetaMask Version: 10.8.1

**Comment by victormel0** [comment]
Same problem. Anyone know how to fix it?

**Comment by Einoow** [comment]
> Same problem. Anyone know how to fix it?

Still waiting, hope they fix it in the next update

**Comment by Zenitsu227** [comment]
Experiencing the same issue.

Browser details (please complete the following information):
OS: Windows
Hardware Wallet: Trezor model one Firmware version 1.10.5
Browser: Firefox
MetaMask Version: 10.8.1

**Comment by Smileez1234** [comment]
same problem as above...also had this problem a few weeks back and tried reinstalling firmware on trezor multiple times and it just started working.  cant confirm that firmware was the problem as I tried this multiple times before having any success.  this problem just started again last night.  has anyone found a solution?

**Comment by Smileez1234** [comment]
I also tried restoring the wallet from a new device (metamask no problem) but when importing the hardware wallet from metamask, it brings up a list of wallets to import, none of which is the address I am looking for.

**Comment by ryny24** [comment]
This issue is so annoying. Happens almost every day. Is this being resolved?

**Comment by Einoow** [comment]
I have checked the new update and it is still happening to me

**Comment by PabloCastellano** [comment]
Still happening to me in 10.9.2

**Comment by Relic180** [comment]
Also having the issue. 

OS: Windows
Hardware Wallet: Trezor model T Firmware version 2.4.3
Browser: Chrome
MetaMask Version: 10.9.2

Is there anything else helpful people could provide to help fix this issue? A way to collect logs or something that we could attach here?

**Comment by AlexJupiter** [comment]
Thanks for your comments. @bitblock, @Smileez1234, @ryny24, @PabloCastellano are you also using Windows OS here? I'm trying to figure out the commonalities.

**Comment by PabloCastellano** [comment]
@AlexJupiter Nope, I'm using Linux Mint. Not sure if it's a commonality but when I have hit this issue I was using the Polygon network, and it worked for other networks intead.

**Comment by PabloCastellano** [comment]
In my case I cannot get to swap on Polygon.

Could you try the following use cases? This is what I'm seeing:

In uniswap (https://app.uniswap.org/#/swap?chain=polygon), the prices keeps updating all the time and I cannot click "Confirm" to initiate the swap tx.

In 1inch (https://app.1inch.io/#/137/swap/USDT/MATIC): when I want to approve the smart contract for the token I want to trade, a metamask popup appears and I click to confirm signing the transaction, but metamask (and firefox at some sort) freezes.

**Comment by Relic180** [comment]
Are there any details on this? Is it something that is actually going to get fixed?

I just recently added my Trezor T to Metamask and turns out that probably +80% of the actions I try to take are locking up the wallet, INCLUDING sending funds that I transferred from my hotwallet, back to my hotwallet in an attempt to just get unblocked.

Recently tried to execute a swap on bogged.finance, and after who knows how long (only noticed it the next day), the transaction failed with the message "Transaction decoding is not available for chainId 56" in the MM transaction data.

**Comment by AlexJupiter** [comment]
@Relic180 thanks for the note, I'm working towards prioritising this very soon. Apologies for the delay but thank you again for your comments.

**Comment by AlexJupiter** [comment]
For any MetaMask engineers looking at this issue please also see this thread related to a similar issue reported by Brave https://consensys.slack.com/archives/C010AA832TX/p1645636789546909 and I have also been in touch with an institutional customer with a similar issue. This _could_ be related to the passphrase feature in Trezor.

**Comment by Relic180** [comment]
Any updates? Looks like new version of MM was released (10.10.2) but issue is still there. Also, a new version of the Trezor Suite was released (22.2.3) although I forgot to check my version before the update. Firmware is the same, from last November.

**Comment by Relic180** [comment]
Another update from both MM (10.11.1) and Trezor Suite (22.3.2), same behavior... all my assets that were sent to Hardware wallet address are stuck.

Is anybody even looking into this issue? Is progress being tracked in another bug maybe?

**Comment by hekreates** [comment]
OS: Windows
Hardware Wallet: Trezor Model One - Firmware version 1.10.5
Browser: Chrome
MetaMask Version: 10.11.1
Everything updated.

Same issue here. Is anyone on the team working to fix it?

**Comment by Relic180** [comment]
Updated to MM 10.11.3, same issue.

**Comment by derekn1** [comment]
Exactly the same issue as above.  

Make the confirm, all freezes, nothing more happens - need to reload browser - transaction not even showing as failed on MM.

OS: Windows
Hardware Wallet: Trezor Model One - Firmware version 2.4.3
Browser: Firefox
MetaMask Version: 10.11.3
Everything updated.

Worrying all metamask coins are completely stuck.

Please advise on update or any workaround.

Thanks

**Comment by derekn1** [comment]
Hi - Update - I was using the MM pass phrase thinking it would confirm, wrong, I used the passphrase on the Trezor and all fine!

**Comment by Relic180** [comment]
I tried several times entering the passphrase directly on the Trezor device for several different types of transactions (sending through MM, approving a dapp, swapping via DEX), but I'm still having exactly same behavior: MM is frozen for a long while (+20 minutes) and eventually times out.

**Comment by Relic180** [comment]
Realized after the last time I let the extension actually timeout (rather than killing/restarting it) that I had access to a log of the failed transaction. The stack trace for one of the errors was:

```Error: Error: Unknown address\n  at m._signTransaction (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:291507)\n  at async G.signTransaction (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:229299)\n  at async G.approveTransaction (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:227597)\n  at async G.updateAndApproveTransaction (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:226988)\n  at async a.<anonymous> (chrome-extension://nkbihfbeogaeaoehlefnkodbefgpgknn/background-0.js:1:150051)```

This was for a `"type": "approve"`, but again I have the same behavior for any transaction I try.

**Comment by seaona** [comment]
Same. After sending a transaction and accepting it on Trezor device, Metamask is frozen. I see this screen:
![image](https://user-images.githubusercontent.com/54408225/163351689-59e0e364-301c-42f6-b7de-d8135f70af6c.png)

After some minutes, Tx is failed. I see the error: `unknown address`
![image](https://user-images.githubusercontent.com/54408225/163351916-15a7f90e-7382-4630-a643-70f1129abcd6.png)

Steps:
1. Load a new build 10.13.0
2. Unlock Trezor
3. Create account in MM
4. Import 3 Trezor addresses
5. Make tx
6. Accept on Trezor

![image](https://user-images.githubusercontent.com/54408225/163352557-b06d88a6-416a-4cb3-982d-132084347ddf.png)

I believe this has to do with the previous actions I did. Even though this was a fresh build, on the previous build I wanted to Import Trezor accounts and got the error `Manifest not set`. See [this issue](https://github.com/MetaMask/metamask-extension/issues/10359) for more info.
After that I logged out, remove the build and start a new one, with the result explained above.
I suspect Trezor stayed on a certain state, leading to this behaviour, as Metamask was a new instance from scratch everytime.

**Comment by seaona** [comment]
to add extra info, I've encountered a similar situation, with Ledger [here](https://github.com/MetaMask/metamask-extension/issues/14426)

**Comment by danjm** [comment]
The freeze is happening when the for loop in _pathFromAddress in eth-trezor-keyring is hit

@seaona will post a recording

**Comment by seaona** [comment]
https://user-images.githubusercontent.com/54408225/163396415-49a5451d-53e5-4c0f-9fc6-ef994a5185b6.mp4

**Comment by EarthSalamander42** [comment]
Having same issue on ETH main network, moved my funds from MM address to Trezor imported into MM for extra safety, now i can't move them anymore, same freeze than everyone else here. Also tried importing MM + trezor to another browser but i can't find the same ETH adresses that i have my funds into. This is kinda scary, i'm wondering what will happen to my funds if my windows crash or my HDD dies in the meantime of a fix being found...

**Comment by EarthSalamander42** [comment]
Looks like when i've imported Trezor wallets, the addresses are not the right ones. Because if i import it in 2 new browsers i have the same addresses in both, but the ones initially imported into Brave where my funds are locked are different. Could that be it?

**Comment by EarthSalamander42** [comment]
Looks like when i've imported Trezor wallets, the addresses are not the right ones. Because if i import it in 2 new browsers i have the same addresses in both, but the ones initially imported into Brave where my funds are locked are different. Could that be it?

**Comment by seaona** [comment]
Here is the complete explanation of what is happening.

**What is causing the issue**: 
The problem is caused due to mixing Trezor addresses that are being imported with a different passphrase.
Basically, when you import Trezor addresses you are asked to provide a passphrase (see screen below). This passhprase dictates the addresses that are being displayed and can be imported.
In the case you import Trezor addresses with one passphrase, lock Metamask and then import Trezor addresses with another passphrase, the first ones will be un-usable and Metamask will freeze in the loop indicated on the comment above.

![image](https://user-images.githubusercontent.com/54408225/166714181-6569094d-38a4-4a9f-8353-b33359329442.png)

**Reproduction steps:**
1. Import a couple of Trezor addresses with a keyphrase. I.e. "123"
2. Lock Metamask
3. Unlock Metamask -- Trezor addresses should be there
4. Import a couple of Trezor addresses - now using a different keyphrase. I.e. "456"
5. Now try to use one address from step 1 - Metamask will freeze

https://user-images.githubusercontent.com/54408225/166713910-088b4249-8671-4ed7-9c6f-b08896eef01a.mp4

**How to solve the issue (temporary)**
First of all, you must identify which is the passphrase that you want to use (with the derived addresses). Review which are the passphrases that you have used in the past (it will be more than one if you are encountering this issue).

Once you know it, you can simply remove Metamask and install it again, and import Trezor addresses using that specific passphrase.
In the case you want to use addresses from a different passphrase I would recommend to use another Metamask instance in another browser and use the other passphrase, but never mix them.

A fix will be develop soon, so this situation does not happen.

**Comment by Tantos1** [comment]
I didnt use a passphrase and when I try on another PC without the passphrase Im still unable to derive the address that I need

**Comment by sime** [comment]
@Tantos1 since you are not using a passphrase, could you try and disable passphrase in Trezor Suite settings and go through the same steps again.

I wonder if having passphrase enabled is apart of the problem.

**Comment by Tantos1** [comment]
I tried that as well it also did not work

**Comment by aliceturino** [comment]
Is there any prediction in solving this issue? I can't use this temporary solution and my money is stuck.

**Comment by yuki-js** [comment]
I also found the root cause and finally discovered this issue.

To avoid freeze, make sure that your address is exactly same as the one Trezor Suite shows, and never type incorrect passphrase when unlocking Trezor (leave it blank if you don't have passphrase) . If you have different wallets with different passphrase, please unplug it.
When you got into freeze, wait for 1000-times-of-heavy-computation, or terminate JS thread. Some browser such as Firefox need force restart.

**Comment by onkaraman** [comment]
Is this issue resolved? I am currently experiencing something very familiar.
